pallets/gear-eth-bridge/src/lib.rs

```diff
@@ -310,16 +333,48 @@ pub mod pallet {
             origin: OriginFor<T>,
             destination: H160,
             payload: Vec<u8>,
-        ) -> DispatchResultWithPostInfo {
-            let source = ensure_signed(origin)?.cast();
+        ) -> DispatchResultWithPostInfo
+        where
+            T::AccountId: Origin,
+        {
+            let source: ActorId = ensure_signed(origin.clone())?.cast();
+
+            Self::try_transfer_fee(&T::AccountId::from_origin(source.into()))?;
 
             Self::queue_message(source, destination, payload)?;
 
             Ok(().into())
         }
+
+        /// Root extrinsic that sets fee for the transport of messages.
+        #[pallet::call_index(3)]
+        #[pallet::weight(<T as Config>::WeightInfo::send_eth_message())]
+        pub fn set_fee(origin: OriginFor<T>, fee: BalanceOf<T>) -> DispatchResultWithPostInfo {
+            // Ensuring called by `AdminOrigin` or root.
+            T::AdminOrigin::ensure_origin_or_root(origin)?;
+
+            // Setting the fee.
+            TransportFee::<T>::put(fee);
+
+            // Returning successful result without weight refund.
+            Ok(().into())
+        }
     }
 
     impl<T: Config> Pallet<T> {
+        fn try_transfer_fee(origin: &T::AccountId) -> DispatchResult
```

Comment:
feel free to extract this code directly into extrinsic body, It won't be reused anywhere

Reply:
This function exists not for reusability, but for logical separation. Is there any reason this could be a bad approach?

Reply:
Fixed